test_runner/regress/test_timeline_delete.py

```diff
@@ -369,30 +380,41 @@ def first_call_hit_failpoint():
 
         wait_until(50, 0.1, first_call_hit_failpoint)
 
-        # make the second call and assert behavior
-        log.info("second call start")
-        error_msg_re = "another task is already setting the deleted_flag, started at"
-        with pytest.raises(PageserverApiException, match=error_msg_re) as second_call_err:
-            ps_http.timeline_delete(env.initial_tenant, child_timeline_id)
-        assert second_call_err.value.status_code == 500
-        env.pageserver.allowed_errors.append(f".*{child_timeline_id}.*{error_msg_re}.*")
-        # the second call will try to transition the timeline into Stopping state as well
-        env.pageserver.allowed_errors.append(
-            f".*{child_timeline_id}.*Ignoring new state, equal to the existing one: Stopping"
+        barrier = threading.Barrier(2)
+        second_call_thread = threading.Thread(
+            target=delete_timeline_call,
+            args=(
+                "2nd",
+                delete_results,
+                barrier,
+            ),
         )
-        log.info("second call failed as expected")
+        second_call_thread.start()
+
+        barrier.wait()
 
-        # by now we know that the second call failed, let's ensure the first call will finish
+        # release the pause
         ps_http.configure_failpoints((failpoint_name, "off"))
 
-        result = first_call_result.get()
+        # both should had succeeded
+        result = delete_results.get()
+        assert result == "success"
+        result = delete_results.get()
         assert result == "success"
 
+        # the second call will try to transition the timeline into Stopping state, but it's already in that state
+        env.pageserver.allowed_errors.append(
+            f".*{child_timeline_id}.*Ignoring new state, equal to the existing one: Stopping"
+        )
```

Comment:
Would it make sense to assert that this log entry is present, in addition to allowing it?
I think so, because, if we ever fix the Tenant / Timeline state transitions to prevent `Stopping->Stopping` "transitions", the log message should be gone and we could remove the allow from this test.

Reply:
I wonder if it could be racy but perhaps this test was the one with a pause, if there was a pause, then it should be an assert.

Reply:
b4cf3b18